lib/active_merchant/billing/gateways/vantiv_express.rb

```diff
@@ -0,0 +1,475 @@
+            add_payment_method(xml, payment)
+            add_transaction(xml, money, options, eci)
+            add_terminal(xml, options, eci)
+            add_address(xml, options)
+            add_lodging(xml, options)
+          end
+        end
+
+        commit(request, money, payment)
+      end
+
+      def authorize(money, payment, options = {})
+        eci = payment.is_a?(NetworkTokenizationCreditCard) ? parse_eci(payment) : nil
+
+        request = build_xml_request do |xml|
+          xml.CreditCardAuthorization(xmlns: live_url) do
+            add_credentials(xml)
+            add_payment_method(xml, payment)
+            add_transaction(xml, money, options, eci)
+            add_terminal(xml, options, eci)
+            add_address(xml, options)
+            add_lodging(xml, options)
+          end
+        end
+
+        commit(request, money, payment)
+      end
+
+      def capture(money, authorization, options = {})
+        trans_id, _, eci = authorization.split('|')
+        options[:trans_id] = trans_id
+
+        request = build_xml_request do |xml|
+          xml.CreditCardAuthorizationCompletion(xmlns: live_url) do
+            add_credentials(xml)
+            add_transaction(xml, money, options, eci)
+            add_terminal(xml, options, eci)
+          end
+        end
+
+        commit(request, money)
+      end
+
+      def refund(money, authorization, options = {})
+        trans_id, _, eci = authorization.split('|')
+        options[:trans_id] = trans_id
+
+        request = build_xml_request do |xml|
+          xml.CreditCardReturn(xmlns: live_url) do
+            add_credentials(xml)
+            add_transaction(xml, money, options, eci)
+            add_terminal(xml, options, eci)
+          end
+        end
+
+        commit(request, money)
+      end
+
+      def credit(money, payment, options = {})
+        eci = payment.is_a?(NetworkTokenizationCreditCard) ? parse_eci(payment) : nil
+
+        request = build_xml_request do |xml|
+          xml.CreditCardCredit(xmlns: live_url) do
+            add_credentials(xml)
+            add_payment_method(xml, payment)
+            add_transaction(xml, money, options, eci)
+            add_terminal(xml, options, eci)
+          end
+        end
+
+        commit(request, money, payment)
+      end
+
+      def void(authorization, options = {})
+        trans_id, trans_amount, eci = authorization.split('|')
+        options.merge!({ trans_id: trans_id, trans_amount: trans_amount, reversal_type: 1 })
+
+        request = build_xml_request do |xml|
+          xml.CreditCardReversal(xmlns: live_url) do
+            add_credentials(xml)
+            add_transaction(xml, trans_amount, options, eci)
+            add_terminal(xml, options, eci)
+          end
+        end
+
+        commit(request, trans_amount)
+      end
+
+      def store(payment, options = {})
+        request = build_xml_request do |xml|
+          xml.PaymentAccountCreate(xmlns: SERVICE_LIVE_URL) do
```

Comment:
How is the `service_live_url` different than `live_url` and is there a reason for `store` to be the only one referencing `service_live_url`?

Reply:
the service_live_url is `services.elementexpress.com` and the live_url is `transaction.elementexpress.com` They have a bunch of service methods and transaction methods for their api, and store is the only method that we have that hits their services api.

Reply:
This is also how the element (non-xml) gateway works.